float32 centre
float32 width
float32 height